### ISSUE TITLE
Question about content adversarial loss L^{content}_{adv}

### ISSUE THREAD

**Issue by lzhbrian** [HsinYingLee/DRIT#17]
Hi, thanks for this marvelous work!
I am reading the paper and found equation (2) a little bit confusing.
Why is the content adversarial loss function like this ?

![codecogseqn 1](https://user-images.githubusercontent.com/12770792/45528357-75cb9b00-b812-11e8-9b72-03f11963ec4a.gif)

If we assign domain x to be 0, and domain y to be 1, then shouldn't this binary cross entropy loss look like? :

![codecogseqn](https://user-images.githubusercontent.com/12770792/45528287-1f5e5c80-b812-11e8-87dc-f5ba7db418ce.gif)

Have I misunderstood anything?
Thanks in advance!

**Comment by lzhbrian** [comment]
Thanks for the quick reply.

I'm still a little confused.

I understand that E's job is to make the features indiscriminative (D(E(x))=1/2)

However, if I understand it right, D^c 's job is to distinguish encoded content features between domain X and domain Y.
Given that we are using one discriminator D^c for both domain X and Y, what D^c would like to do is to make domain X closer to 0, and domain Y closer to 1. (i.e. to make them apart from each other as far as possible)
If these two parts are considered separately, what would D^c's job actually be?

**Comment by zhangqianhui** [comment]
@lzhbrian @HsinYingLee  I also think this loss has some problems.

**Comment by fransilvion** [comment]
I am not even sure if this loss is correctly implemented in the code... it's backward_contentD function from the train model, right? This loss is calculated only once in the code:

`loss_D_Content = self.backward_contentD(self.z_content_a, self.z_content_b)`

And it corresponds to the following loss:

> logDc(Eca(image_a)) + log(1 - Dc(Ecb(image_b))

However, to be equal to equation 2 in the paper, the code should be:

`loss_D_Content1 = self.backward_contentD(self.z_content_a, self.z_content_b)`
`loss_D_Content2 = self.backward_contentD(self.z_content_b, self.z_content_a)`
`loss_D_Content = 0.5*loss_D_Content1 + 0.5*loss_D_content2`

The minimum of this loss is when for both Dc(Eca(image_a)) and Dc(Ecb(image_b)) is 1/2. 
Am I missing something? Please, let me know if I am wrong.

**Comment by controny** [comment]
@lzhbrian 
The loss for D^c and that for E are not the same.
Equation (2) in the paper is the loss for E, according to function [backward_G_GAN_content](https://github.com/HsinYingLee/DRIT/blob/master/src/model.py#L357).
The loss for D^c should be E_x[log(1-D^c(E^c_x(x))] + E_y[logD^c(E^c_y(y))], where label for x is 0 and label for y is 1, according to function [backward_contentD](https://github.com/HsinYingLee/DRIT/blob/master/src/model.py#L274).
